meson.build

```
@@ -0,0 +1,633 @@
+# -*- mode: meson -*-
+
+# Style objective: be consistent with what mesonbuild.com documents/uses, and/or
+# the meson book: https://meson-manual.com/
+
+project(
+  'i3',
+  'c',
+  version: '4.18.1',
+  default_options: [
+    'c_std=c11',
+    'warning_level=1',  # enable all warnings (-Wall)
+    # TODO(https://github.com/i3/i3/issues/4087): switch to
+    # 'buildtype=debugoptimized',
+  ],
+  # TODO: we can target more recent versions of meson starting with i3 v4.20,
+  # at which point Ubuntu 20.04 is reasonable to expect.
+  meson_version: '>=0.45.0',
```

But why not keep support for older ones as well? 18.04 has standard support until 2023. Not saying that it is our duty to keep i3 working on ancient software but historically i3 has been much more conservative than this.

I've tried and this works:

```suggestion
  meson_version: '>=0.44.0',
```

Maybe we should make sure in our CI that we are also testing with the minimal supported meson version.

---

Okay, fair enough. I have softened the TODO, let’s see how this goes. We can revisit if it becomes a burden.

Why the suggestion for 0.44.0? Is there a popular Linux version that doesn’t have 0.45.0?

---

The suggestion was more like a FYI. Ideally we would like to support the minimal supported version. Practically, using something that we can have on our CI is less of a burden.

---

Let’s target 0.45.0 then. I added a TODO item to https://github.com/i3/i3/issues/4086 for this.